dashboard/app/views/admin_reports/retention.html.haml

```diff
@@ -7,6 +7,13 @@
 
 %br
 
+%form{method: 'get'}
+  Script IDs:
+  %input{name: "script_ids", id: "script_ids", value: params[:script_ids]}
```

Comment:
if you want to keep the default behavior that you are adding here without modifying params you can do `value: @script_ids` (where @script_ids is a string you build in the controller) or just directly build it here with something like `value: @scripts.map(&:id).join(", ")`
it would not be too hard to make this a multiselect that can be prefilled based on the value of @scripts (and using script names instead of/in addition to the ids) with the rails form tag helpers.